stone/frontend/ast.py

```diff
@@ -344,10 +344,11 @@ def __repr__(self):
 
 class AstRouteDef(ASTNode):
 
-    def __init__(self, path, lineno, lexpos, name, deprecated,
+    def __init__(self, path, lineno, lexpos, name, version, deprecated,
                  arg_type_ref, result_type_ref, error_type_ref=None):
         super(AstRouteDef, self).__init__(path, lineno, lexpos)
         self.name = name
```

Comment:
If we add a version number as part of the canonical name, then we are not able to detect the name conflict between a route and an item of a different type (struct, namespace, etc.). There will be a test error at `test_stone.py:848`.

Reply:
Can we add another function to check the route name without version? The original _check_canonical_name_available is still helpful to prevent version collision.